### PR TITLE
Chore: remove the uncessary allAssociatedRefGrants from resourceMappings

### PR DESCRIPTION
**Pull request by zhaohuabing**
This PR removes `allAssociatedRefGrants` from `resourceMappings` and directly add `ReferenceGrant` to `Resources`.

### CODE REVIEW

File: site/content/en/latest/user/security/oidc.md [hunk 1]
```diff
@@ -16,13 +16,42 @@ This instantiated resource can be linked to a [Gateway][Gateway] and [HTTPRoute]
 Follow the steps from the [Quickstart](../../quickstart) guide to install Envoy Gateway and the example manifest.
```
Comment (zirain) [inline site/content/en/latest/user/security/oidc.md:16]: split this change to a separate PR, with a more suitable title?

Comment (zhaohuabing) [reply site/content/en/latest/user/security/oidc.md:16]: This file has been modifed by mistake, I'll remove it.

**Review by arkodg — APPROVED** [review]
LGTM thanks

**Review by shawnh2 — APPROVED** [review]
